src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -1028,6 +1029,134 @@ private TwoWindingsTransformer createTwoWindingsTransformer(Network network, Vol
+                Integer rank1 = idToNodeRank.get(fromBBSId);
+                if (rank1 == null) {
+                    throw new NetworkModificationException(CREATE_VOLTAGE_LEVEL_ERROR, "From side '" + fromBBSId + "' unknown");
+                }
+
+                String toBBSId = bbsci.getToBBS();
+                Integer rank2 = idToNodeRank.get(toBBSId);
+                if (rank2 == null) {
+                    throw new NetworkModificationException(CREATE_VOLTAGE_LEVEL_ERROR, "To side '" + toBBSId + "' unknown");
+                }
+
+                SwitchKind switchKind = bbsci.getSwitchKind();
+                if (switchKind == SwitchKind.BREAKER) {
+                    int preBreakerRank = nodeRank++;
+                    int postBreakerRank = nodeRank++;
+                    voltageLevel.getNodeBreakerView().newDisconnector()
+                        .setKind(SwitchKind.DISCONNECTOR)
+                        .setId("switch_" + voltageLevelCreationInfos.getEquipmentId() + "_" + fromBBSId + "_" + toBBSId
+                            + "_" + cnxRank++)
+                        .setNode1(rank1)
+                        .setNode2(preBreakerRank)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+
+                    voltageLevel.getNodeBreakerView().newBreaker()
+                        .setKind(switchKind)
+                        .setId("breaker_" + voltageLevelCreationInfos.getEquipmentId() + "_" + cnxRank++)
+                        .setNode1(preBreakerRank)
+                        .setNode2(postBreakerRank)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+
+                    voltageLevel.getNodeBreakerView().newDisconnector()
+                        .setKind(SwitchKind.DISCONNECTOR)
+                        .setId("switch_" + voltageLevelCreationInfos.getEquipmentId() + "_" + toBBSId + "_" + fromBBSId
+                            + "_" + cnxRank++)
+                        .setNode1(postBreakerRank)
+                        .setNode2(rank2)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+                } else {
+                    voltageLevel.getNodeBreakerView().newDisconnector()
+                        .setKind(switchKind)
+                        .setId("switch_" + voltageLevelCreationInfos.getEquipmentId() + "_" + fromBBSId + "_" + toBBSId
+                            + "_" + cnxRank++)
+                        .setNode1(rank1)
+                        .setNode2(rank2)
+                        .setFictitious(false)
+                        .setRetained(false)
+                        .setOpen(false)
+                        .add();
+                }
```

Comment:
Maybe throw an exception if SwitchKind is not BREAKER or DISCONNECTOR ?

Reply:
done